src/Language/Syntax.hs

```diff
@@ -1,12 +1,15 @@
+{-# LANGUAGE DeriveDataTypeable #-}
 -- | Abstract syntax for BOGL
 
 module Language.Syntax where
 import Data.List
+import Text.JSON.Generic
 import qualified Data.Set as S
 type Name = String
 
 -- | Game datatype
 data Game = Game Name BoardDef InputDef [ValDef]
+  deriving (Data)
```

Comment:
Something that can be done about these is the extension StandaloneDeriving, which lets you derive typeclasses in modules other than where the data type was defined. With that, all the JSON-related stuff could be in just one file. This works just fine, though, so it's up to you.